homeassistant/components/workday/config_flow.py

```diff
@@ -0,0 +1,291 @@
+
+    for add_date in user_input[CONF_ADD_HOLIDAYS]:
+        if dt.parse_date(add_date) is None:
+            raise AddDatesError("Incorrect date")
+
+    year: int = dt.now().year
+    obj_holidays: HolidayBase = getattr(holidays, user_input[CONF_COUNTRY])(years=year)
+    if user_input.get(CONF_PROVINCE):
+        obj_holidays = getattr(holidays, user_input[CONF_COUNTRY])(
+            subdiv=user_input[CONF_PROVINCE], years=year
+        )
+
+    for remove_date in user_input[CONF_REMOVE_HOLIDAYS]:
+        if dt.parse_date(remove_date) is None:
+            if obj_holidays.get_named(remove_date) == []:
+                raise RemoveDatesError("Incorrect date or name")
+
+
+DATA_SCHEMA_SETUP = vol.Schema(
+    {
+        vol.Required(CONF_NAME, default=DEFAULT_NAME): TextSelector(),
+        vol.Required(CONF_COUNTRY): SelectSelector(
+            SelectSelectorConfig(
+                options=list(holidays.list_supported_countries()),
+                mode=SelectSelectorMode.DROPDOWN,
+            )
+        ),
+    }
+)
+
+DATA_SCHEMA_OPT = vol.Schema(
+    {
+        vol.Optional(CONF_EXCLUDES, default=DEFAULT_EXCLUDES): SelectSelector(
+            SelectSelectorConfig(
+                options=ALLOWED_DAYS, multiple=True, mode=SelectSelectorMode.DROPDOWN
+            )
+        ),
+        vol.Optional(CONF_OFFSET, default=DEFAULT_OFFSET): NumberSelector(
+            NumberSelectorConfig(min=-10, max=10, step=1, mode=NumberSelectorMode.BOX)
+        ),
+        vol.Optional(CONF_WORKDAYS, default=DEFAULT_WORKDAYS): SelectSelector(
+            SelectSelectorConfig(
+                options=ALLOWED_DAYS, multiple=True, mode=SelectSelectorMode.DROPDOWN
+            )
+        ),
+        vol.Optional(CONF_ADD_HOLIDAYS, default=[]): SelectSelector(
+            SelectSelectorConfig(
+                options=[],
+                multiple=True,
+                custom_value=True,
+                mode=SelectSelectorMode.DROPDOWN,
+            )
+        ),
+        vol.Optional(CONF_REMOVE_HOLIDAYS, default=[]): SelectSelector(
+            SelectSelectorConfig(
+                options=[],
+                multiple=True,
+                custom_value=True,
+                mode=SelectSelectorMode.DROPDOWN,
+            )
+        ),
+    }
+)
+
+
+class WorkdayConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Workday integration."""
+
+    VERSION = 1
+
+    data: dict[str, Any] = {}
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> WorkdayOptionsFlowHandler:
+        """Get the options flow for this handler."""
+        return WorkdayOptionsFlowHandler(config_entry)
+
+    async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
+        """Import a configuration from config.yaml."""
+
+        input_config = {
+            k: v for k, v in config.items() if k not in [CONF_NAME, CONF_PLATFORM]
```

Comment:
I think you should use an `in` instead, to be more explicit.
```suggestion
            k: v for k, v in config.items() if k in {...}
```